File: solutions/r/difference-of-squares/1/difference-of-squares.R
difference_of_squares <- function(natural_number) {
  # Sum of first n natural numbers: n * (n + 1) / 2
  sum_numbers <- natural_number * (natural_number + 1) / 2
  
  # Square of the sum
  square_of_sum <- sum_numbers^2
  
  # Sum of squares of first n natural numbers: n * (n + 1) * (2n + 1) / 6
  sum_of_squares <- natural_number * (natural_number + 1) * (2 * natural_number + 1) / 6
  
  # Return the difference
  square_of_sum - sum_of_squares
}

File: solutions/go/difference-of-squares/1/difference_of_squares.go
package differenceofsquares

func SquareOfSum(n int) int {
	// Sum of first n natural numbers: n * (n + 1) / 2
	sum := n * (n + 1) / 2
	return sum * sum
}

func SumOfSquares(n int) int {
	// Sum of squares of first n natural numbers: n * (n + 1) * (2n + 1) / 6
	return n * (n + 1) * (2*n + 1) / 6
}

func Difference(n int) int {
	return SquareOfSum(n) - SumOfSquares(n)
}